src/OpenTelemetry.Exporter.Geneva/Internal/AFDCorrelationIdLogProcessor.cs

```diff
@@ -42,6 +50,37 @@ public override void OnEnd(LogRecord data)
         base.OnEnd(data);
     }
 
+    protected override void Dispose(bool disposing)
+    {
+        if (!this.disposed)
+        {
+            if (disposing)
+            {
+                GenevaAfdCorrelationIdStateTracker.Dispose();
+            }
+
+            this.disposed = true;
+        }
+
+        base.Dispose(disposing);
+    }
+
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    private static string? GetRuntimeContextValue() => RuntimeContext.GetValue<string>(AFDCorrelationId);
+    private static string? GetRuntimeContextValue()
+    {
+        if (GenevaAfdCorrelationIdStateTracker.Get() == true)
+        {
+            return null;
+        }
+
+        try
+        {
+            return RuntimeContext.GetValue<string>(AFDCorrelationId);
+        }
+        catch
+        {
+            GenevaAfdCorrelationIdStateTracker.Set(true);
```

Comment:
nit: it's not clear what `true` and `false` mean for "StateTracker".
To improve readability consider naming this something like "IsGenevaAfdCorrelationIdAvailable"

Reply:
If there is an issue, the customer could set this from their app. Want to avoid "Is" in the name.